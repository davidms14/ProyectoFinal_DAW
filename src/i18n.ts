import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

/**
 * Este documento contiene las funciones para poder implementar la internalización en el proyecto 
 */

    //Idiomas a implementar
// English
import en from './assets/locales/en.json';

// Spanish
import es from './assets/locales/es.json';

export enum Language {
    English = 'en',
    Spanish = 'es'
}

//Array de palabras traducidas en el documento
const cldr = {
    [Language.English]: new Array(),
    [Language.Spanish]: new Array(),
};

const resources = {
    [Language.English]: {
        translation: en
    },
    [Language.Spanish]: {
        translation: es
    }
};

i18n
    .use(initReactI18next) // pasa i18n abajo a react-i18next
    .init({
        resources,
        lng: Language.English, // Cambia el lenguaje actual al final del documento
        interpolation: {
            escapeValue: false 
        },
    });


export default i18n;

export function getCurrentLanguage(): string {
    return i18n.language;
}

export type LanguageReturnType = string[];

export function getLanguagesList(): LanguageReturnType[] {
    return [
        ['English', Language.English],
        ['Español', Language.Spanish]
    ];
}

export function changeCurrentLanguage(language: Language): void {
    i18n.changeLanguage(language);

}

// Si no esta implementado el inglés por defecto, usamos el español
changeCurrentLanguage(Language.Spanish);