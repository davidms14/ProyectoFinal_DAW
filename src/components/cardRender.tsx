import '../index.css';
import '../style.css';
import * as React from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { getCurrentLanguage } from '../i18n';
import moment from 'moment';

/**
 * @param cardRender - Esta clase funciona como un componente para Tile ya que es la encargada 
 * de contener la infomación que se mostrará en las tarjetas, los datos de los clientes, fechas, muestras
 * de estas... Además tendrá unos iconos que nos permitirá conocer en todo momento el estado de las tarjetas
 * que estamos mostrando. Con estos iconos que ejercen las funciones de botón podremos visualizar un reporte
 * de manera individual, observar las muestras enlazadas a dicho reporte y descargar el reporte seleccionado.
 */

export interface Data {
    id: string;
    reportNumber: string;
    client: string;
    entryDate: Date;
    expectedDate: Date;
    status: string;
    reference: string;
    certificate: string;
}

interface Props {
    data: Data;
}

interface CardRenderState {
    data: Data;
    isLoadSamples: boolean;
    hasMore: boolean;
}

export type CardRenderPropsWithTranslation = Props & WithTranslation;

export class CardRender extends React.Component<CardRenderPropsWithTranslation, CardRenderState> {
    public constructor(props: CardRenderPropsWithTranslation) {
        super(props);

        this.state = {
            data: this.props.data,
            isLoadSamples: false,
            hasMore: true,
        };
    }

    /**
     * 
     * @returns {string} Devuelve string
     * @example
     * //Example funcion
     * @description
     * //Con esta función podemos controlar el estado y saber si
     * se ha acabado de trabajar en un reporte o sigue en proceso. Esto puede saberse grácias a los iconos
     * que nos ayudan a identificarlo.
     * @param {string} status - Hace refencia al estado del reporte.
     * 
     */
    private getStatus(status: string): string {
        if (status === "finalized") {
            /** Tarea finalizada */
            return status = "icon-checkbox-checked iconosTarjetas";
        }
        else {
            /** Tarea por finalizar */
            return status = "icon-hour-glass iconosTarjetas"
        }
    }

    /**
     * @returns {string} Devuelve string
     * @example
     * //Ejemplo funcion
     * @description
     * //Esta función simplemente permite saber el significado del icono que
     * nos indica el estado del reporte. Así al pasar el ratón por encima de este podemos saber 
     * que significa.
     * @param {string} title - Hace refencia al estado del reporte.
     */

    private getTitle(title: string): string {
        if (title === "finalized") {
            return title = this.props.i18n.t('status.finalized');
        }
        else {
            return title = this.props.i18n.t('status.inprogress');
        }
    }

    /**
     * @example
     * //Example funcion
     * @description
     * //La función se encarga de cargar las muestras, también modifica el estado de estas
     * ya que no siempre sacará las mismas, sino que deberá sacar la muestras enlazadas al reporte.
     * @param {boolean} isLoadSample - Controla si las muestras se han cargado
     */

    private getAlert(isLoadSample: boolean): void {
        /*var newState = this.state.data;*/
        this.setState({ /*data: newState,*/ isLoadSamples: isLoadSample });
    }


    public render(): JSX.Element {
        /** Almacena el contenido de la tarjeta, se mostrará información de la tarjeta o las muestras */
        let content = [];


        /** Antes se tomaban las muestras desde otro documento, ahora al ser un campo dentro del actual JSON accedemos a dicho campo
         * y lo metemos en un array. En cada posición se inserta una muestra, estas se separan por comas.
         */
        let references: string[] = this.state.data.reference.split(', ');
        let notEmptyReferences: string[] = references.filter((element: string): boolean => {
            return element !== undefined && element != null && element.trim() !== '';
        });

        /** Este icono nos mostrará las muestras, si estamos en ellas desaparece. */
        const buttonToShowSamples =
            (notEmptyReferences.length > 0 && !this.state.isLoadSamples) ?
                <i title={this.props.i18n.t('icons.samples')} className="icon-file-empty iconosTarjetas"
                    key={'buttonToShowSamples_'.concat(this.state.data.id)}
                    onClick={this.getAlert.bind(this, true)}
                >
                </i>
                :
                null;

        /** Con ese icono podremos visualizar el reporte */
        const buttonToShow =
            this.state.data.certificate === "true" ?
                <i title={this.props.i18n.t('icons.detail')} className="icon-link iconosTarjetas" ></i>
                :
                null;

        /** Con ese icono podremos descargar el reporte */
        const buttonToDownload =
            this.state.data.status === "finalized" ?
                <i title={this.props.i18n.t('icons.download')} className="icon-download3 iconosTarjetas" ></i>
                :
                null;

        /** Si estamos viendo las muestras, para volver a los datos del reporte pulsaremos aqui. */
        const buttonToClose =
            <i title={this.props.i18n.t('icons.cerrar')} className="icon-cross iconoCerrar"
                key={'buttonToCloseSamples_'.concat(this.state.data.id)}
                onClick={this.getAlert.bind(this, false)}
            ></i>;

        /** Contiene los datos a mostrar en la tarjeta */
        if (!this.state.isLoadSamples) {
            if (this.state.data.status === "finalized") {
                moment.locale(getCurrentLanguage());

                content.push(

                    <div key={'div_report_'.concat(this.state.data.id)}>
                        <b title={this.props.i18n.t('home.filters.client')}>{this.props.i18n.t('home.filters.client')}:</b> {this.state.data.client}
                        <br />
                        <b title={this.props.i18n.t('home.filters.entry')}>{this.props.i18n.t('home.filters.finish')}:</b> {moment(this.state.data.entryDate).format('L')}
                        <br />
                        <b title={this.props.i18n.t('home.filters.expected')}>{this.props.i18n.t('home.filters.timed')}:</b> {moment(this.state.data.expectedDate).add(5, 'years').format('L')}
                    </div>
                );
            } else {
                moment.locale(getCurrentLanguage());

                content.push(

                    <div key={'div_report_'.concat(this.state.data.id)}>
                        <b title={this.props.i18n.t('home.filters.client')}>{this.props.i18n.t('home.filters.client')}:</b> {this.state.data.client}
                        <br />
                        <b title={this.props.i18n.t('home.filters.entry')}>{this.props.i18n.t('home.filters.entry')}:</b> {moment(this.state.data.entryDate).format('L')}
                        <br />
                        <b title={this.props.i18n.t('home.filters.expected')}>{this.props.i18n.t('home.filters.expected')}:</b> {moment(this.state.data.expectedDate).format('L')}
                    </div>
                );
            }
        }

        /** Muestras los reportes asociados a la tarjeta*/
        else if (this.state.data.reference.trim() !== '') {
            content.push(buttonToClose);
            references.forEach((element: string, index: number): void => {
                content.push(
                    <div key={'div_reference_'.concat(index.toString())}>
                        <div><b>-</b> {element}</div>
                    </div>
                );
            });
        }

        /** Devuelve la esctrutura de la tarjeta, esta consta de un HEADER donde encontramos el estado y el número de reporte. 
         * Ambas cosas siempre estarán en ese sitio.
         * 
         * Seguidamente, tenemos el apartado de CONTENT, el cual si que varia dependiendo de si estamos mostrando la información
         * de la tarjeta o las muestras.
         * 
         * Para acabar tendriamos un apartado de ACTIONS, este contiene los tres botones de detalles, descarga y muestras y al igual que la cabecera
         * estos siempre ocuparán ese lugar.
          */
        return (
            <div className='e-card'>

                <div className='e-card-header'>
                    <div className='e-card-header-title' title={this.props.i18n.t('home.table.reportNumber')}>
                        <i title={this.getTitle(this.state.data.status)}
                            className={this.getStatus(this.state.data.status)}>
                        </i>{this.state.data.reportNumber}
                    </div>
                </div>

                <div className='e-card-content' style={{ overflowY: 'auto' }}>
                    {content}
                </div>

                <div className='e-card-actions iconos'>
                    {buttonToShow}
                    {buttonToShowSamples}
                    {buttonToDownload}
                </div>
            </div>
        );
    }
}


export default withTranslation()(CardRender);


