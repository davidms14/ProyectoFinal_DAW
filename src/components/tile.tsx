import * as React from 'react';
import InfiniteScroll from 'react-infinite-scroll-component';
import { serverData as serverAllData } from '../data-source';
import { Data } from './cardRender';
import CardRender from './cardRender';
import { withTranslation, WithTranslation } from 'react-i18next';

/**
 * @param Tile - El Tile contiene todos los reportes alojados a su vez en un contenedor.
 * Está conectada con el componente de cardRender.
 * 
 * Primero encontramos las funciones para tomar los datos de las tarjetas de lo que sería 
 * el servidor y la función para cargar las "páginas" con un número determinado de
 * componentes.
 * 
 * Tras el render que será lo que muestre la vista de nuestro componente, encontramos 
 * los valores que hacen referecia al componente del Scroll, este nos permite cargar datos
 * automáticamente sin la necesidad de estar continuamente pulsando un botón para cargar más reportes.
 * 
 * Para finalizar tenemos la estructura del contenedor que almacenará las tarjetas con sus respectivos
 * datos.
 */

export interface State {
    /** Cantidad e Informes almacenados */
    reports: Data[];
    /** Si quedan informes por ser almacenados (están en el servidor) */
    hasMore: boolean;
}

export class Tile extends React.PureComponent<WithTranslation, State> {

    /** Con ella se controla cual va a ser la página en la cual se empezarán a incluir reportes. */
    public nextPage: number = 0;
    /** Se encarga de almacenar los datos que vienen del serverData para poder trabajar con ellos. */
    public cardBook: any = serverAllData;

    public constructor(props: any) {
        super(props);

        this.fetchMoreData = this.fetchMoreData.bind(this);

        this.state = {
            reports: [],
            hasMore: true
        };
    }

    /**
     * @returns {Function} Funcion
     * @example
     * // Ejemplo funcion
     * @description
     * // Simula la obtención de datos del Servidor.
     * @param {number} skip - La página que se quiere cargar (se empieza por 0) 
     * @param {number} top - El numero de tarjetas que se mostraran por cada página.
     * 
     */
    private getCardsFromServer(skip: number, top: number): Data[] {
        let length: number = this.cardBook["@odata.count"];
        let startIndex: number = (skip) * top;
        let endIndex: number = startIndex + top;

        if (endIndex <= length + top) {
            return this.cardBook.value.slice(startIndex, endIndex);
        }
        else {
            return [];
        }

    }

    /** 
     * @example
     * //Example function
     * @description
     * // Esta función se encarga de colocar X reportes (los que nosotros queramos mostrar en cada página).
     * Una vez mostrados los primeros, cambiará de página y llenará esta nuevamente con el numero de 
     * reportes que se haya acordado.
     */
    private cargaPags(): void {
        let newCards: Data[] = this.getCardsFromServer(this.nextPage, 8);
        if (newCards.length === 0) {
            this.setState({ hasMore: false });
        } else {
            this.nextPage++;
            let showedCard: Data[] = [];
            this.state.reports.forEach((cardData): void => {
                showedCard.push(cardData);
            });
            newCards.forEach((cardData): void => {
                //TODO: revisar que ese valor no esté ya en el array
                showedCard.push(cardData);
            });
            this.setState({ reports: showedCard });
        }
    }


    /**
     * @returns {Function} Funcion
     * @example
     * //Example function
     * @description
     * // Esta función se encarga de controlar si quedan más reportes por mostrar, si no quedan,
     * no se puede seguir haciendo scroll.
     */
    public fetchMoreData(): void {
        if (this.state.reports.length >= this.cardBook.value.length) {
            this.setState({ hasMore: false });
        }
        else {
            this.cargaPags();
        }
    };

    public componentDidMount(): void {
        /**
         * Si la página a mostrar no tiene elementos, volverá a la función de la página para cargar datos 
         */
        if (this.state.reports.length === 0) {
            this.cargaPags();
        }
    }

    public render(): JSX.Element {

        /** Prepara las propiedades que tendrá el componente de scroll infinito. Se encarga de controlar cuantos elementos se van a mostrar
         * el tamaño que tendra el componente, cada cuanto tiene que mostrar elementos, que mensaje mostrar cuando esté todo cargado...
         */
        const config = {
            dataLength: this.state.reports.length,
            next: this.fetchMoreData,
            hasMore: this.state.hasMore,
            height: 900,
            loader: "",
            endMessage:
                <div className="divEndMessage">
                    <b className="endMessage">Todos los elementos cargados</b>
                </div>
        };

        /** Aquí se controla si hay reportes en el array, en caso de que si, crea la tarjeta, le asigna una ID y la maqueta. */
        let myReports: JSX.Element[] = [];


        if (this.state.reports.length > 0) {
            this.state.reports.forEach((cardData): void => {
                let myCardEle = <div key={'card_sample_'.concat(cardData.id)} id={'card_sample_' + cardData.id} className="controlTarjetas">
                    <CardRender key={'CardRender_'.concat(cardData.id)} data={cardData} />
                </div>;
                myReports.push(myCardEle);
            });
        }

        /** Prepara el contenedor donde irán alojadas los reportes, estas se mostrarán en 4 columnas. El infinity scroll hace referencia
         * al componente que nos permite que solo carge los elementos haciendo scroll sobre el elemento ROW, no sobre la página entera (window).
         */
        return (
            <div className='control-pane'>
                <div className="control-section card-control-section tile_layout">
                    <div className=' e-card-layout ' >
                        <InfiniteScroll {...config}>
                            <div className="row rowStyle">
                                <div id="target">
                                    {myReports}
                                </div>
                            </div>
                            <br />
                            <div ><button id="boton" hidden={!this.state.hasMore} onClick={this.cargaPags.bind(this, "")} className="newCard" >Cargar reportes</button></div>
                        </InfiniteScroll>
                    </div>
                </div>
            </div>
        );
    }
}

export default withTranslation()(Tile);

