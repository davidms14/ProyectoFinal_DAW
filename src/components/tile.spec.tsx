import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { Tile, State } from './tile';
import { Data } from './cardRender';
import { WithTranslation } from 'react-i18next';
import i18n from '../i18n';

/**
 * Documento para el testo de componente, en este caso para el código de Tile.tsx
 */

/**
  * Primer test: test de interfaz o visual, enfocado a comprobar la parte del código que se ve en el navegador mediante un snapshot
  */
 
describe('components/tile tests', () => {
    it('should render as expected when passing required properties', () => {
        // Arrange
            //Aquí se realiza la toma de datos con los que trabajaremos en este test
        const props: WithTranslation = {
            i18n: i18n,
        } as WithTranslation;

        // Act
            //Indicamos sobre que funciones y/o apartados trabajará la prueba
        const component = mount(
            <Tile {...props} />
        );

        // Assert
            //Aquí indicamos el resultado que queremos obtener, en este caso un snapshot
        expect(component.debug()).toMatchSnapshot();
    });

    /**
     * Segundo test: test para comprobar si una vez se han cargado todos los elementos, el programa deja de cargarlos.
     */
    it('should stop charging pages', () => {

        // Arrange
        const props: WithTranslation = {
            i18n: i18n,
        } as WithTranslation;

        // Act
        const component = mount(
            <Tile {...props} />
        );

        // Assert

        component.find('.newCard').simulate('click');
        debugger;
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');
        component.find('InfiniteScroll').simulate('scroll');

    });

    /**
     * Tercer test: test para comprobar que se cargan todos los datos.
     */
    it('should enter in cargaPags', () => {

        // Arrange
        const props: WithTranslation = {
            i18n: i18n,
        } as WithTranslation;

        // Act
        const component = mount(
            <Tile {...props} />
        );

        // Assert
        component.find('.newCard').simulate('click');
        debugger;

    });
});