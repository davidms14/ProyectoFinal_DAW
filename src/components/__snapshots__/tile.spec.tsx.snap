// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/tile tests should render as expected when passing required properties 1`] = `
"<Tile i18n={{...}}>
  <div className=\\"control-pane\\">
    <div className=\\"control-section card-control-section tile_layout\\">
      <div className=\\" e-card-layout \\">
        <InfiniteScroll dataLength={8} next={[Function: bound fetchMoreData]} hasMore={true} height={900} loader=\\"\\" endMessage={{...}} pullDownToRefreshContent={{...}} releaseToRefreshContent={{...}} pullDownToRefreshThreshold={100} disableBrowserPullToRefresh={true}>
          <div style={{...}}>
            <div className=\\"infinite-scroll-component \\" style={{...}}>
              <div className=\\"row rowStyle\\">
                <div id=\\"target\\">
                  <div id=\\"card_sample_6fdd1b8f-b0b4-4b40-931f-fb95bb7e9b9a\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"En Proceso\\" className=\\"icon-hour-glass iconosTarjetas\\" />
                              ReporteReporte21
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte1
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha Inicio
                                :
                              </b>
                               
                              09/04/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha estimada
                                :
                              </b>
                               
                              09/04/2019
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Ir a detalle\\" className=\\"icon-link iconosTarjetas\\" />
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                  <div id=\\"card_sample_75e4a24e-b5a3-4349-a562-d343f5a1c6e9\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"Finalizado\\" className=\\"icon-checkbox-checked iconosTarjetas\\" />
                              Reporte2
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte2
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha de finalización
                                :
                              </b>
                               
                              24/01/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha de caducidad
                                :
                              </b>
                               
                              04/02/2024
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                            <i title=\\"Descargar\\" className=\\"icon-download3 iconosTarjetas\\" />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                  <div id=\\"card_sample_7c790df6-b2bd-4539-a4b5-17c88ab4d517\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"Finalizado\\" className=\\"icon-checkbox-checked iconosTarjetas\\" />
                              Reporte3
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte3
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha de finalización
                                :
                              </b>
                               
                              24/01/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha de caducidad
                                :
                              </b>
                               
                              04/02/2024
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Ir a detalle\\" className=\\"icon-link iconosTarjetas\\" />
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                            <i title=\\"Descargar\\" className=\\"icon-download3 iconosTarjetas\\" />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                  <div id=\\"card_sample_a8d1029a-a7c1-4b24-9840-19e95cc56fcd\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"Finalizado\\" className=\\"icon-checkbox-checked iconosTarjetas\\" />
                              Reporte4
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte4
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha de finalización
                                :
                              </b>
                               
                              24/01/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha de caducidad
                                :
                              </b>
                               
                              04/02/2024
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Ir a detalle\\" className=\\"icon-link iconosTarjetas\\" />
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                            <i title=\\"Descargar\\" className=\\"icon-download3 iconosTarjetas\\" />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                  <div id=\\"card_sample_f5ae40b6-f2d2-40fb-a62f-6070806a043d\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"Finalizado\\" className=\\"icon-checkbox-checked iconosTarjetas\\" />
                              Reporte5
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte5
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha de finalización
                                :
                              </b>
                               
                              24/01/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha de caducidad
                                :
                              </b>
                               
                              07/02/2024
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Ir a detalle\\" className=\\"icon-link iconosTarjetas\\" />
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                            <i title=\\"Descargar\\" className=\\"icon-download3 iconosTarjetas\\" />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                  <div id=\\"card_sample_f7f47a21-352f-43ce-9ae5-8c0dca111481\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"Finalizado\\" className=\\"icon-checkbox-checked iconosTarjetas\\" />
                              Reporte6
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte6
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha de finalización
                                :
                              </b>
                               
                              11/01/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha de caducidad
                                :
                              </b>
                               
                              22/02/2024
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Ir a detalle\\" className=\\"icon-link iconosTarjetas\\" />
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                            <i title=\\"Descargar\\" className=\\"icon-download3 iconosTarjetas\\" />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                  <div id=\\"card_sample_7fdd1b8f-b0b4-4b40-931f-fb95bb7e9b9a\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"En Proceso\\" className=\\"icon-hour-glass iconosTarjetas\\" />
                              Reporte7
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte7
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha Inicio
                                :
                              </b>
                               
                              09/04/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha estimada
                                :
                              </b>
                               
                              09/04/2019
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Ir a detalle\\" className=\\"icon-link iconosTarjetas\\" />
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                  <div id=\\"card_sample_8fdd1b8f-b0b4-4b40-931f-fb95bb7e9b9a\\" className=\\"controlTarjetas\\">
                    <withI18nextTranslation(CardRender) data={{...}}>
                      <CardRender data={{...}} t={[Function: fixedT]} i18n={{...}} tReady={true}>
                        <div className=\\"e-card\\">
                          <div className=\\"e-card-header\\">
                            <div className=\\"e-card-header-title\\" title=\\"Nº Informe\\">
                              <i title=\\"En Proceso\\" className=\\"icon-hour-glass iconosTarjetas\\" />
                              Reporte8
                            </div>
                          </div>
                          <div className=\\"e-card-content\\" style={{...}}>
                            <div>
                              <b title=\\"Cliente\\">
                                Cliente
                                :
                              </b>
                               
                              Reporte8
                              <br />
                              <b title=\\"Fecha Inicio\\">
                                Fecha Inicio
                                :
                              </b>
                               
                              09/04/2019
                              <br />
                              <b title=\\"Fecha estimada\\">
                                Fecha estimada
                                :
                              </b>
                               
                              09/04/2019
                            </div>
                          </div>
                          <div className=\\"e-card-actions iconos\\">
                            <i title=\\"Mostrar muestras\\" className=\\"icon-file-empty iconosTarjetas\\" onClick={[Function: bound getAlert]} />
                          </div>
                        </div>
                      </CardRender>
                    </withI18nextTranslation(CardRender)>
                  </div>
                </div>
              </div>
              <br />
              <div>
                <button id=\\"boton\\" hidden={false} onClick={[Function: bound cargaPags]} className=\\"newCard\\">
                  Cargar reportes
                </button>
              </div>
            </div>
          </div>
        </InfiniteScroll>
      </div>
    </div>
  </div>
</Tile>"
`;
